tempest/api/workloadmgr/restore/test_exclude_cinder_volume.py

```diff
@@ -0,0 +1,266 @@
+            LOG.debug("Assigned floating IP : " + str(floating_ip_1))
+            time.sleep(20)
+
+            #add some data to volume attached
+            mount_point = ["mount_data_AFolder"]
+            self.data_ops_on_mount_point(floating_ip_1, mount_point[0], 3)
+            LOG.debug("Created disk and mounted the attached volume")
+
+            #add metadata to the volume...
+            custom_metadata = {"exclude_from_backup": "True"
+                        }
+
+            body = self.volumes_client.update_volume_metadata(
+            volume_id, custom_metadata)['metadata']
+
+            LOG.debug("Requested to update metadata: " + str(body))
+
+            # Create workload using CLI
+            workload_create = command_argument_string.workload_create + \
+                                              " --instance instance-id=" + str(vm_id)
+
+            rc = cli_parser.cli_returncode(workload_create)
+
+            if rc != 0:
+                reporting.add_test_step(
+                    "Execute workload-create command using cli",
+                    tvaultconf.FAIL)
+                raise Exception(
+                    "Command workload create did not execute correctly")
+            else:
+                reporting.add_test_step(
+                    "Execute workload-create command using cli",
+                    tvaultconf.PASS)
+
+            time.sleep(10)
+
+            #check workload status
+            wid = query_data.get_workload_id_in_creation(tvaultconf.workload_name)
+            LOG.debug("Workload ID: " + str(wid))
+            if(wid is not None):
+                self.wait_for_workload_tobe_available(wid)
+                if(self.getWorkloadStatus(wid) == "available"):
+                    reporting.add_test_step(
+                        "Create workload using cli", tvaultconf.PASS)
+                else:
+                    reporting.add_test_step(
+                        "Create workload using cli", tvaultconf.FAIL)
+                    raise Exception("Create workload using cli failed.")
+            else:
+                reporting.add_test_step(
+                        "Create workload using cli", tvaultconf.FAIL)
+                raise Exception("Create workload using cli failed.")
+
+
+            #take full snapshot
+            snapshot_id = self.workload_snapshot(wid, True)
+
+            self.wait_for_workload_tobe_available(wid)
+            snapshot_status = self.getSnapshotStatus(wid, snapshot_id)
+
+            if (snapshot_status == "available"):
+                LOG.debug("Full snapshot created.")
+                reporting.add_test_step("Create full snapshot", tvaultconf.PASS)
+                tests[0][1] = 1
+            else:
+                reporting.add_test_step("Create full snapshot", tvaultconf.FAIL)
+                raise Exception("Full snapshot creation failed.")
+
+            reporting.test_case_to_write()
```

Comment:
Why is snapshot creation considered a separate test case ? Restore is the test case here I suppose.